Test/Classes/TestTextUtils.cs

```diff
@@ -36,13 +36,121 @@ public class TestTextUtils
-            Assert.Equal("Test", TextUtils.StripDateFromName("Test 13 February 2009", date));
-            Assert.Equal("Test", TextUtils.StripDateFromName("Test 13th Feb 2009", date));
-            Assert.Equal("Test", TextUtils.StripDateFromName("Test 13th February 2009", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13-2-09", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13-2-2009", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13-02-09", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13-02-2009", date));
+
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13/2/09", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13/2/2009", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13/02/09", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13/02/2009", date));
+
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13 Feb 09", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13 Feb '09", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13 Feb 2009", date));
+
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13-Feb-09", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13-Feb-2009", date));
+
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13 February 09", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13 February '09", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13 February 2009", date));
+
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13th Feb 2009", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 13th February 2009", date));
+
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 2009-02-13", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 2009/02/13", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test 2009.02.13", date));
+
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test Feb 13th 09", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test Feb 13th '09", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test Feb 13th 2009", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test Feb 13 09", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test Feb 13 '09", date));
+                Assert.Equal("Test", TextUtils.StripDateFromName("Test Feb 13 2009", date));
+            }
+            catch
+            {
+                Assert.False(true);
+            }
```

Comment:
Again, if there is an exception we should fail with it instead of catching it (to make debugging easier).